consultation_analyser/consultations/views/answers.py

```diff
@@ -144,98 +145,90 @@ def respondents_json(
     cache_key = f"respondents_{request.user.id}_{consultation_slug}_{question_slug}"
     cache_timeout = 60 * 20  #  20 mins
 
-    responseid = request.GET.get("responseid")
-    demographicindividual = request.GET.getlist("demographicindividual")
-    themesfilter = request.GET.getlist("themesfilter")
-    themesentiment = request.GET.get("themesentiment")
-    responsesentiment = request.GET.get("responsesentiment")
-    wordcount = request.GET.get("wordcount")
-
+    # Retrieve cached data
     data = cache.get(cache_key)
 
+    # If no cached data found
     if data is None:
-        # Get question data
-        consultation = get_object_or_404(models.Consultation, slug=consultation_slug)
-
-        question = models.Question.objects.get(
-            slug=question_slug,
-            consultation=consultation,
+        # Prefetch all related data to avoid multiple db hits
+        filtered_answers = models.Answer.objects.filter(
+            question_part__question__slug=question_slug
+        ).prefetch_related(
+            Prefetch("thememapping_set", to_attr="prefetched_thememappings")
+        ).prefetch_related(
+            Prefetch("evidencerichmapping_set", to_attr="prefetched_evidencerichmappings")
+        ).prefetch_related(
+            Prefetch("sentimentmapping_set", to_attr="prefetched_sentimentmappings")
         )
-        free_text_question_part = models.QuestionPart.objects.filter(
-            question=question, type=models.QuestionPart.QuestionType.FREE_TEXT
-        ).first()  # Assume that there is only one free text response
-
-        # Get respondents list, applying relevant filters
-        respondents = filter_by_response_and_theme(
-            question,
-            models.Respondent.objects.filter(consultation=consultation),
-            responseid,
-            responsesentiment,
-            themesfilter,
-            themesentiment,
-        ).distinct()
 
-        if wordcount:
-            respondents = filter_by_word_count(respondents, question_slug, int(wordcount))
-
-        has_individual_data, respondents = filter_by_demographic_data(
-            respondents, demographicindividual
-        )
+        respondents = models.Respondent.objects.filter(
+            consultation__slug=consultation_slug
+        ).prefetch_related(
+            Prefetch("answer_set", queryset=filtered_answers, to_attr="prefetched_answers")
+        ).distinct()
 
-        # Get summary data for filtered respondents list
-        selected_theme_mappings, theme_mapping_summary = get_selected_theme_summary(
-            free_text_question_part, respondents
-        )
+        data = {
+            "all_respondents": []
+        }
 
         # Get individual data for each displayed respondent
         for respondent in respondents:
+            
```

Comment:
This might be an investigation for a future PR, but I wonder if we can remove this loop if - instead of prefetching all answers as `prefetched_answers` - we filter the filtered_answer queryset by question_part.type (FREE_TEXT or not) and record these separately, e.g. as `prefetched_free_text_answers` and `prefetched_multiple_choice`.

It'll need some work on updating the references for the template, but might get the data there quicker.

Reply:
happy to leave this to a spike for another ticket (as discussed in Slack)